source/features/stop-redirecting-in-notifications-bar.tsx

```diff
@@ -0,0 +1,25 @@
+import delegate from 'delegate-it';
+import features from '../libs/features';
+import * as pageDetect from '../libs/page-detect';
+
+function onNotificationActionClick(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
+	const shouldRedirectToInbox = !event.altKey;
+	const actionForm = event.delegateTarget.closest('form')!;
+	actionForm.toggleAttribute('data-redirect-to-inbox-on-submit', shouldRedirectToInbox);
+}
+
+async function init(): Promise<void> {
+	const actionButtonSelector = '.notifications-v2.notification-shelf .js-notification-action button';
+	delegate(document, actionButtonSelector, 'click', onNotificationActionClick);
+}
+
+features.add({
+	id: __filebasename,
+	description: 'Stops redirecting to notification inbox from notification bar actions while holding `Alt`.',
+	screenshot: 'https://user-images.githubusercontent.com/202916/80318782-c38cef80-880c-11ea-9226-72c585f42a51.png'
+}, {
+	include: [
+		pageDetect.isRepo
```

Comment:
I think
```suggestion
		pageDetect.isPR,
		pageDetect.isIssue
```

Reply:
At least commits and releases can be notification targets, too.

Reply:
Indeed. I then we need a new detection function in the current file. Maybe:

```js
const hasNotification = () => location.search.startsWith('?notification_referrer_id=');
```

And then:

```js
{
	include: [
		hasNotification
	],
	waitForDomReady: false,
	repeatOnAjax: false,
	init
}
```

Reply:
This won't work if you reload on a page with notification bar, as the URL query parameter is removed.

E.g. `https://github.com/sindresorhus/refined-github/pull/3036?notification_referrer_id=XXXXXXXX#pullrequestreview-401122761` will be detected. That turns to `https://github.com/sindresorhus/refined-github/pull/3036#pullrequestreview-401122761` immediately, so after a reload, that page will not be detected, but the notifications bar is still there.

Reply:
> so after a reload [...] the notifications bar is still there.

That doesn't match my experience:

![demo2](https://user-images.githubusercontent.com/1402241/80547681-1e634a00-89b9-11ea-830a-e9176b9c01a9.gif)

From what I see, GitHub uses exactly the same `location.search` test

Reply:
Weird, when I reload this page, the notifications bar re-appears after a short delay. I also tried disabling all GitHub-related browser extensions (Refined GitHub, OctoLinker and Notifications Preview for GitHub), but my experience doesn't change.

Reply:
> It apparently only re-appears on PRs, not issues or releases.

Indeed. Weirdly enough, PRs store this state in `sessionStorage`. Maybe:

```js
const hasNotificationBar = () =>
	location.search.startsWith('?notification_referrer_id=') ||
	JSON.parse(sessionStorage.notification_shelf ?? '{}').pathname === location.pathname;
```

Reply:
That works fine!